verification-annotations/src/klee.rs

```diff
@@ -60,6 +62,51 @@ make_verifier_nondet!(isize);
 make_verifier_nondet!(f32);
 make_verifier_nondet!(f64);
 
+/// Allocate a symbolic vector of bytes
+pub fn verifier_nondet_bytes(n: usize) -> Vec<u8> {
+    // create an empty vector with capacity 'n'
+    let v: Vec<u8> = Vec::with_capacity(n);
+
+    // take vector apart so that we can access unused capacity
+    let mut v = mem::ManuallyDrop::new(v);
+    let p = v.as_mut_ptr();
+
+    unsafe {
+        // mark contents of vector as symbolic
+        let data = p as *mut raw::c_void;
+        let null = 0 as *const i8;
+        klee_make_symbolic(data, n, null);
+
+        // put vector back together again,
+        // with same capacity but increased length
+        Vec::from_raw_parts(p, n, n)
+    }
+}
+
+/// Allocate a symbolic CString
+pub fn verifier_nondet_cstring(size_excluding_null: usize) -> CString {
+    let r = verifier_nondet_bytes(size_excluding_null + 1);
+    for i in 0..size_excluding_null {
+        assume(r[i] != 0u8);
+    }
+    assume(r[size_excluding_null] == 0u8);
```

Comment:
Sure, I'll do that.

I think I was thinking about how to work with immutable objects - for which the assume is the right choice.
But it is easy to make it mutable in this case since I had only just created it.